dataeng/jobs/AnalyticsEmailOptinMaster.groovy

```diff
@@ -0,0 +1,105 @@
+job ('analytics-email-optin-master-dsl-test') {
+
+    parameters{
+        stringParam('ORGS','*', 'Space separated list of organizations to process. Can use wildcards. e.g.: idbx HarvardX')
+        stringParam('EXPORTER_BRANCH','environment/production',
+                'Branch from the edx-analytics-exporter repository. For tags use tags/[tag-name]. Should be environment/production.')
+        stringParam('PLATFORM_BRANCH','zafft/analytics-exporter-settings-hotfix',
+                'Branch from the edx-platform repository. For tags use tags/[tag-name]. Should be release.')
+        stringParam('SECURE_BRANCH','release',
+                'Branch from the analytics-secure repository, where the configuration settings reside. For tags use tags/[tag-name]')
+        stringParam('CONFIG_FILENAME','default.yaml', 'Name of configuration file in analytics-secure/analytics-exporter.')
+        stringParam('OUTPUT_BUCKET','course-data/email-opt-in', 'Name of the bucket for the destination of the email opt-in data.')
+        stringParam('OUTPUT_PREFIX','email-opt-in-', 'Optional prefix to prepend to output filename.')
+        stringParam('NOTIFICATION_EMAILS','analytics@edx.opsgenie.net analytics-alerts@edx.org',
+                'Space separated list of emails to notify in case of failure.')
+        stringParam('DATE_MODIFIER','',
+                'Used to set the date of the CWSM dump.  Leave blank to use today\'s date.  Set to "-d 201x-0x-0x" if that is when the CWSM dump took place, typically the preceding Sunday.  (Leave off quotes.)')
+        stringParam('ORG_CONFIG','data-czar-keys/config.yaml', 'Path to the data-czar organization config file')
+        stringParam('DATA_CZAR_KEYS_BRANCH','master', 'Branch of the Data-czar-keys repository to use')
+    }
+
+    multiscm{
+        git {
+            remote {
+                url('git@github.com:edx/edx-platform.git')
+                branch('$PLATFORM_BRANCH')
+                credentials('1')
+            }
+            extensions {
+                pruneBranches()
+                relativeTargetDirectory('edx-platform')
+            }
+        }
+        git {
+            remote {
+                url('https://github.com/edx/edx-analytics-exporter.git')
+                branch('$EXPORTER_BRANCH')
+            }
+            extensions {
+                pruneBranches()
+                relativeTargetDirectory('analytics-exporter')
```

Comment:
My ignorance: is there a reason this repo doesn't have a "credentials('1')" call here, and all others do?

Reply:
So the current Email Optin job uses https for this repo(I went with what we have), we can skip the credentials while using https. If we use 'git@github.com' url, we need to provide credentials.

Reply:
We are safe to continue using https as we don't have the TLS 1.0, TLS 1.1 issue on the new box.  However we may need to switch to SSH to take advantage of the SSH pem key in the global credentials.